src/Interpreters/Cache/FileCacheFactory.cpp

```diff
@@ -155,7 +155,17 @@ void FileCacheFactory::updateSettingsFromConfig(const Poco::Util::AbstractConfig
 
         FileCacheSettings old_settings = cache_info->getSettings();
         if (old_settings == new_settings)
+        {
             continue;
+        }
+
+        /// FIXME: registerDiskCache modifies `path` setting of FileCacheSettings if path is relative.
+        /// This can lead to calling applySettingsIfPossible even though nothing changed, which is avoidable.
+
+        // LOG_TRACE(log, "Will apply settings changes for cache {}. "
+        //           "Settings changes: {} (new settings: {}, old_settings: {})",
+        //           cache_name, fmt::join(new_settings.getSettingsDiff(old_settings), ", "),
```

Comment:
Do we need `getSettingsDiff` then?

Reply:
This comment is good and I like to have it, but it is not correct to log it unless a "fixme" above is fixed. So I do not know, may be we can delete `getSettingsDiff` now and revive it once the fixme is fixed, or do not delete it now and let it just be. I do not have a strong opinion here. Do you think better to delete?